### PR TITLE
refactor(ReportsPageRanking): change service method

### PR DESCRIPTION
**Pull request by fcoronelmakingsense**
- add withEmail field for visitors with email
- fix test
- change service method to manage errors

### CODE REVIEW

**Comment by andresmoschini** [comment]
❤️

File: src/services/datahub-client.ts [hunk 1]
```diff
@@ -123,23 +132,37 @@ export class HttpDatahubClient implements DatahubClient {
...
+        {
+          startDate: dateFrom.toISOString(),
+          sortBy: 'visitors',
+        },
+      );
 
-    // By the moment we are hard-coding it because DataHub does not have this
-    // information. I am looking you Leo :P
-    const urlSchema = 'http://';
+      // By the moment we are hard-coding it because DataHub does not have this
```
Comment (cbernat) [inline src/services/datahub-client.ts]: I think this comment does not apply anymore right?

Comment (fcoronelmakingsense) [reply src/services/datahub-client.ts]: we dont have the schema because datahub dont provide us, then need to use this for now :(

Comment (andresmoschini) [reply src/services/datahub-client.ts]: Related ticket: [DBR-271](http://jira.makingsense.com/browse/DBR-271)

**Comment by cbernat** [comment]
🚀

**Pull request merged**